### ISSUE TITLE
[bug]: Mocking of DialogService throws error after upgrade to v1.1.0

### ISSUE THREAD

**Issue by sebastianbuechler** [Stacked-Org/stacked#977]
### Describe the bug

I just ugpraded to `stacked-services` v1.1.0 (together with `stacked` v3.4.0) and now the build runner fails when I want to generate the mocks for the `DialogService` with the `mockito` package.

### To reproduce

1. Install packages (  stacked: ^3.4.0,  stacked_services: ^1.1.0, build_runner: ^2.1.4, mockito: ^5.4.0)
2. Define the `DialogService` to have mocks generated: `@GenerateNiceMocks( [   MockSpec<DialogService>(),]);`
3. Let build runner run and get error output (and no mock file):
    [SEVERE] mockito:mockBuilder on test/setup/test_helpers.dart:
    Null check operator used on a null value

### Expected behavior

Generate the mocks as usual.

**Comment by ferrarafer** [comment]
Hi @sebastianbuechler ! I just created a project from scratch and I could not reproduce your issue.

Could you try to change the DialogService mock spec like below?

```dart
MockSpec<DialogService>(onMissingStub: OnMissingStub.returnDefault),
```

Also, could you update the following packages?

- build_runner: ^2.4.5
- mockito: ^5.4.1

Let me know what happened, I will be around.

**Comment by sebastianbuechler** [comment]
> Hi @sebastianbuechler ! I just created a project from scratch and I could not reproduce your issue.
> 
> Could you try to change the DialogService mock spec like below?
> 
> ```dart
> MockSpec<DialogService>(onMissingStub: OnMissingStub.returnDefault),
> ```
> 
> Also, could you update the following packages?
> 
> * build_runner: ^2.4.5
> * mockito: ^5.4.1
> 
> Let me know what happened, I will be around.

HI @ferrarafer, thanks for the reply. 

I tried to update the `build_runner` to the version you mentioned but I'm still on Dart SDK 2.19.5:
```
flutter pub get
Running "flutter pub get" in MobileApp...
Resolving dependencies...
The current Dart SDK version is 2.19.5.

Because mobileapp depends on build_runner >=2.4.0 which requires SDK version >=3.0.0-134.0.dev <4.0.0, version solving failed.

```
Is the upgrade of the Dart SDK needed for this to work? I'm just asking because it seems a lot of packages are not ready for Dart SDK >= 3.0.0 and intl >= 0.18.0.

Oh, and regarding the MockSpec you mentioned: Isn't that exactly the same when I use the `@GenerateNiceMocks` annotation or did I understand the difference between `GenerateMocks` and `@GenerateNiceMocks` wrongly?

**Comment by sebastianbuechler** [comment]
Update: Managed to upgrade to Dart SDK 3.0.5 and set build_runner to 2.4.5 and mockito 5.4.1. Still the same issue
```
[INFO] Generating build script completed, took 507ms
[INFO] Reading cached asset graph completed, took 6.5s
[INFO] Checking for updates since last build completed, took 2.0s
[WARNING] No actions completed for 23.6s, waiting on:
  - stacked_generator:stackedRouterGenerator on test/setup/test_helpers.dart

[WARNING] No actions completed for 38.8s, waiting on:

[SEVERE] mockito:mockBuilder on test/setup/test_helpers.dart:

Null check operator used on a null value
[INFO] Running build completed, took 1m 23s
[INFO] Caching finalized dependency graph completed, took 5.0s
[SEVERE] Failed after 1m 28s
make: *** [makefile:12: run-build-runner] Error 1
```

I'm using the `flutter pub run build_runner build --delete-conflicting-outputs` command for generating the mocks. Also, same result when using `MockSpec<DialogService>(onMissingStub: OnMissingStub.returnDefault),`.

**Comment by ferrarafer** [comment]
Interesting, I will try to reproduce this on my side.

**Comment by ferrarafer** [comment]
Unfortunately I can't reproduce this issue. What is the possibility that you upload your project to a repository so I can pull it and test it locally? Obviously, it doesn't have to be the complete project, you can leave only the minimum necessary for the issue to be reproducible on your side.

**Comment by sebastianbuechler** [comment]
It's hard to reproduce, as I can't create a new repo that shows the issue. 

However, I did some debugging and found the piece that breaks:
Mockito breaks here: https://github.com/dart-lang/mockito/blob/ffbbb4ce8057fc5f793cc3e2fd635f0903fee3a3/lib/src/builder.dart#L289

And I found that it breaks when handling the `barrierColor` parameter of the `showCustomDialog` of the dialog service of stacked. It seems that mockito has issues with the dart:ui library import, as the type of this `barrierColor` gets reported as `InvalidType`. That's why the line mentioned above probably breaks. 

I saw multiple similar behaviors of mockito mentioned in my ticket here: https://github.com/dart-lang/build/issues/4457

So I don't think it's a stacked issue, but something that mockito does not like about my current project. Interestingly, I tried it on three different machines and all show the same behavior.
